File: benchmarks/java/set/Clojure_Set_From_Java/RangeDifferenceSmall.java
import de.hhu.stups.btypes.BSet;
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import de.hhu.stups.btypes.BUtils;

public class RangeDifferenceSmall {






    private BInteger counter;
    private BSet<BInteger> set;

    public RangeDifferenceSmall() {
        counter = new BInteger(0);
        set = BSet.range(new BInteger(0),new BInteger(5));
    }

    public void simulate() {
        while((counter.less(new BInteger(5000000))).booleanValue()) {
            set = set.difference(BSet.range(new BInteger(1),new BInteger(2)));
            counter = counter.plus(new BInteger(1));
        }
    }

    public static void main(String[] args) {
        RangeDifferenceSmall exec = new RangeDifferenceSmall();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));

        Runtime runtime = Runtime.getRuntime();
        long memory = runtime.totalMemory() - runtime.freeMemory();
        System.out.println("Used memory is bytes: " + memory);
        System.out.println("Used memory is megabytes: "
                + (memory / (1024L * 1024L)));
    }

}

File: benchmarks/cpp/set/RangeDifferenceSmall.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BSet.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"

#ifndef RangeDifferenceSmall_H
#define RangeDifferenceSmall_H

using namespace std;

class RangeDifferenceSmall {



    private:



        BInteger counter;
        BSet<BInteger > set;

    public:

        RangeDifferenceSmall() {
            counter = (BInteger(0));
            set = (BSet<BInteger>::range((BInteger(0)),(BInteger(5))));
        }

        void simulate() {
            while((counter.less((BInteger(5000000)))).booleanValue()) {
                set = set.difference((BSet<BInteger>::range((BInteger(1)),(BInteger(2)))));
                counter = counter.plus((BInteger(1)));
            }
        }

};
int main() {
    clock_t start,finish;
    double time;
    RangeDifferenceSmall exec;
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

